src/plugins/intel_gpu/src/graph/graph_optimizer/reorder_inputs.cpp

```diff
@@ -59,7 +59,19 @@ std::map<program_node*, format::type> get_preferred_formats(program& p, layout_o
             onednn_impls_counter++;
     }
 
-    if (onednn_impls_counter < 1 && lo.get_optimization_attributes().use_onednn_impls) {
+    size_t total_convs = 0;
+    size_t num_wei_zp_convs = 0;
+    for (auto n : p.get_processing_order()) {
+        if (n->is_type<convolution>()) {
+            total_convs++;
+            if (n->as<convolution>().weights_zero_points_term())
+                num_wei_zp_convs++;
+        }
+    }
+
+    GPU_DEBUG_LOG << "Number of convolutions with weights zero points: " << num_wei_zp_convs << "/" << total_convs << std::endl;
+
+    if ((onednn_impls_counter < 1 && lo.get_optimization_attributes().use_onednn_impls) || (num_wei_zp_convs > 0)) {
```

Comment:
Proper condition will be:
((onednn_impls_counter < 1 || (num_wei_zp_convs > 0) && lo.get_optimization_attributes().use_onednn_impls)

Reply:
How is it?
if (lo.get_optimization_attributes().use_onednn_impls && (onednn_impls_counter < 1 || num_asym_wei_convs > 0)) {